### ISSUE TITLE
Cookies for Tweakers.net

### ISSUE THREAD

**Issue by ltGuillaume** [tobykurien/WebApps#175]
I'm trying to create a WebApps entry for https://tweakers.net

Whatever I do, no cookies are saved. I've toggled every setting. After I accept the cookies on the website, the next time I run the WebApps entry, I get the same cookies consent dialog. The same thing happens after I've logged in on the site.

Any way I can get and provide more info on what's going on?

Latest WebApps via F-Droid
Android 9 (recent LineageOS 16) on Galaxy S5
Using Bromite WebView

**Comment by ltGuillaume** [comment]
Logcat entries that may be of concern:
```
07-02 09:18:50.198 I/chatty  ( 1581): uid=10164(com.tobykurien.webapps) identical 24 lines
07-02 09:18:50.200 D/aBatis  ( 1581): Attempt to invoke virtual method 'boolean java.lang.String.equalsIgnoreCase(java.lang.String)' on a null object reference
07-02 09:18:50.655 W/SurfaceFlinger(  498): Attempting to destroy on removed layer: AppWindowToken{e08a2cc token=Token{bd24ff ActivityRecord{654541e u0 com.tobykurien.webapps/.activity.WebAppActivity t13985}}}#0
07-02 09:18:53.552 I/ActivityManager(  840): START u0 {act=android.intent.action.VIEW dat=https://tweakers.net/... cmp=com.tobykurien.webapps/.activity.WebAppActivity (has extras)} from uid 10164
07-02 09:18:53.582 W/ActivityThread( 1581): handleWindowVisibility: no activity for token android.os.BinderProxy@bd9e194
07-02 09:18:53.662 D/aBatis  ( 1581): Attempt to invoke virtual method 'boolean java.lang.String.equalsIgnoreCase(java.lang.String)' on a null object reference
07-02 09:18:53.665 D/aBatis  ( 1581): Attempt to invoke virtual method 'boolean java.lang.String.equalsIgnoreCase(java.lang.String)' on a null object reference

```

**Comment by tobykurien** [comment]
Could you try v3.50 and see if it fixes the issue?

**Comment by ltGuillaume** [comment]
Already running 3.50 via F-Droid and the issue persists.

**Comment by ltGuillaume** [comment]
Cool! Can't wait to test it!!

**Comment by ltGuillaume** [comment]
v4.0 just became available on F-Droid. Unfortunately, I'm getting the exact same results with this version.

**Comment by tobykurien** [comment]
Tweakers.net is really finnicky, yes it doesn't work but if you keep trying, at some point it sticks and works thereafter. It has been a nightmare to test because of this. It seems the session cookie changes on each page load and eventually WebApps will be in sync with the server.

**Comment by ltGuillaume** [comment]
What a b*tchy way to use cookies. I've been trying all day, but the cookie has never "stuck".

**Comment by tobykurien** [comment]
I got it to work like this: save the webapp, open it, accept cookie, close app and reopen it, accept cookies again if it asks and close, and reopen until it doesn't ask anymore. Took me about 5 tries.

**Comment by ltGuillaume** [comment]
Have you blocked any 3rd party cookies or 3rd pary domains? (I'm trying without doing that, too.) I have tried over 40 times by now, the exact way you're describing.

**Comment by tobykurien** [comment]
Yes all 3rd parties were blocked

**Comment by tobykurien** [comment]
I just tried again now after deleting the webapp, and again after the 5th time it stopped asking for cookie acceptance.

**Comment by ltGuillaume** [comment]
There are two factors I can think of that may influence the fact that I can't get it working:
1. I'm using a rather old phone (Samsung Galaxy S5, but with Android 10)
2. I'm using Bromite WebView instead of Android's default WebView.

**Comment by ltGuillaume** [comment]
I just tried on another phone, _not_ rooted, with a clean installation of WebApps via F-Droid. After trying what you described (open - accept -exit) 50 times there still is a cookie wall every time.

**Comment by tobykurien** [comment]
I am testing on a non-rooted Android 9 device, also with F-droid version. Will look into this further when I get a chance.

**Comment by ltGuillaume** [comment]
Any news on this? Anything I can do to get you more info on this?

**Comment by tobykurien** [comment]
No news. It's too tempremental an issue and takes too much effort to debug, so I have given up. Similar to #215 which I am still trying to figure out, and affects me, I was hoping to solve them both in one go, but so far no luck.

**Comment by ltGuillaume** [comment]
Understood. I could inquire about what the hell they're doing with their cookies over there (I'd need some more specific terminology to pose a useful question though 😅), although I don't know if that would also help with the GitHub issue.

**Comment by ltGuillaume** [comment]
I'm getting the same kind of issues with Gitlab now (half-assed logins that show error 422 upon login, but then show myself as logged in, unless I try to reply in an issue), and Gitea instances like Codeberg (I can log in and browse as if logged in properly, but when I post any form I get a session error).

**Comment by tobykurien** [comment]
Yes cookies are saved in onPause() and restored in onResume(). If you click from one webapp to another, all cookies are deleted and then the one for the new webapp restored. Again if you navigate back to previous webapp, all cookies are deleted and saved ones restored. This may have unintended consequences, but is the only way I know to make a good sandbox.

**Comment by ltGuillaume** [comment]
So it would _always_ restore the cookies in onResume()?

But with rolling session tokens or something, that could go wrong quite simply, right? If there's already a cookie, it might have changed since onPause() (or perhaps even right before onResume()), and onResume() will then restore an old cookie without reason.

If, instead, onResume() just checks which app is active and for which app the cookies have last been restored, then it could just not restore anything if the right cookies are already present.

I don't know if _this_ would be the actual issue as described here, but I'd say if I understand the current procedure correctly from you reply, then I think it could unnecessarily introduce problems just like it.

**Comment by tobykurien** [comment]
It only restores cookies if the webapp has changed.

**Comment by ltGuillaume** [comment]
> 
> 
> It only restores cookies if the webapp has changed.

Aah ok, didn't get that from your previous reply, thx 🙂

I guess that whatever's going wrong is totally beyond my comprehension then 😅
